src/connection.rs

```diff
@@ -186,6 +186,88 @@ impl fmt::Debug for Connection {
     }
 }
 
+/// Whether an endpoint was the initiator of a connection
+#[cfg_attr(feature = "arbitrary", derive(Arbitrary))]
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub enum Side {
+    /// The initiator of a connection
+    Initiator = 0,
+    /// The acceptor of a connection
+    Acceptor = 1,
+}
+
+/// Whether a stream communicates data in both directions or only from the initiator
+#[cfg_attr(feature = "arbitrary", derive(Arbitrary))]
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub enum Dir {
+    /// Data flows in both directions
+    Bi = 0,
+    /// Data flows only from the stream's initiator
+    Uni = 1,
+}
+
+impl fmt::Display for Dir {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use self::Dir::*;
+        f.pad(match *self {
+            Bi => "bidirectional",
+            Uni => "unidirectional",
+        })
+    }
+}
+
+/// Identifier for a stream within a particular connection
+#[cfg_attr(feature = "arbitrary", derive(Arbitrary))]
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct StreamId(#[doc(hidden)] pub u64);
+
+impl fmt::Display for StreamId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let initiator = match self.initiator() {
+            Side::Initiator => "initiator",
+            Side::Acceptor => "acceptor",
+        };
+        let dir = match self.dir() {
```

Comment:
You don't need this, or even the `Display` impl for `Dir`, you can just use the `Debug` version of it in L236, i.e. `{dir:?}`.

Reply:
This is for a more concised log for the StreamId, which is different to the default Debug or Dispay impl of `enum Dir`

Reply:
I think you don't need them, let me put an example further below in the other discussion